### PR TITLE
list sca files for a specific ref

### PR DESCRIPTION
**Pull request by gg-mmill**

### CODE REVIEW

File: ggshield/sca/file_selection.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    get_staged_filepaths,
+    tar_from_ref_and_filepaths,
+)
+from ggshield.sca.client import SCAClient
+
+
+def tar_sca_files_from_git_repo(directory: Path, ref: str, client: SCAClient):
+    """Builds a tar containing SCA files from the git repository at directory."""
+    # TODO: add exclusion patterns
+    if ref is None:
```
Comment (tchaumeny) [inline ggshield/sca/file_selection.py:17]: `ref` is typed as `str`, not nullable

File: ggshield/sca/file_selection.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    get_staged_filepaths,
+    tar_from_ref_and_filepaths,
+)
+from ggshield.sca.client import SCAClient
+
+
+def tar_sca_files_from_git_repo(directory: Path, ref: str, client: SCAClient):
+    """Builds a tar containing SCA files from the git repository at directory."""
+    # TODO: add exclusion patterns
+    if ref == "":
```
Comment (tchaumeny) [inline ggshield/sca/file_selection.py:17]: Maybe use a constant as in IaC

**Comment by gg-mmill** [comment]
Added test (and fixed bugs :bug: )

**Review by tchaumeny — APPROVED** [review]
LGTM!

**Pull request merged**